### PR TITLE
Work link

### PR DESCRIPTION
**Pull request by gestchild**
Relates to #8907

Implements the changes related to repositioning and restyling the catalogue link on the item page behind a toggle. The toggle will eventually be used to split traffic into one of two conditions.

Screenshot 2022-12-16 at 13 51 25

Screenshot 2022-12-16 at 13 51 55

N.B there were other styling differences in the Zeplin files, such as background colours, button styles, keylines, which I've ignored as they seem like more general changes to the viewer and not required for the A/B test.

Re the A/B test: we want to test the hypotheses that for everyone that lands on an item page, more people in condition B will click the work page link. We send the toggle data with every page view to Segment, so it's possible to determine how many people over a given period viewed an item page in each condition. We also send a source as a query parameter on the url when the link is clicked.  This is currently 'viewer_back_link' for condition A and, as requested by @taceybadgerbrook, will be 'button_back_link' for condition B. So we will be able to determine over the same period how many times the link was clicked in each condition.

### CODE REVIEW

File: catalogue/webapp/components/IIIFViewer/ViewerSidebar.tsx [hunk 1]
```diff
@@ -143,6 +146,20 @@ const ViewerSidebar: FunctionComponent<Props> = ({ mainViewerRef }: Props) => {
   return (
     <>
       <Inner className={font('intb', 5)}>
+        {itemWorkLink && (
+          <div className="viewer-mobile">
+            <Space
+              v={{ size: 'xl', properties: ['margin-top', 'margin-bottom'] }}
+            >
+              <WorkLink id={work.id} source="button_back_link" passHref>
+                <WorkLinkAnchor className={`${font('intr', 5)}`}>
```
Comment (alexwlchan) [inline catalogue/webapp/components/IIIFViewer/ViewerSidebar.tsx:155]:
```suggestion
                <WorkLinkAnchor className={font('intr', 5)}>
```

File: catalogue/webapp/components/IIIFViewer/ViewerTopBar.tsx [hunk 2]
```diff
@@ -206,16 +224,26 @@ const ViewerTopBar: FunctionComponent<Props> = ({ viewerRef }: Props) => {
...
     >
-      {isEnhanced && (
-        <Sidebar isZooming={showZoomed}>
-          {!showZoomed && (
-            <>
+      <Sidebar isZooming={showZoomed} itemWorkLink={itemWorkLink}>
+        {itemWorkLink && (
+          <div className="viewer-desktop">
+            <WorkLink id={work.id} source="button_back_link" passHref>
+              <WorkLinkAnchor className={`${font('intr', 5)}`}>
```
Comment (alexwlchan) [inline catalogue/webapp/components/IIIFViewer/ViewerTopBar.tsx:238]:
```suggestion
              <WorkLinkAnchor className={font('intr', 5)}>
```

File: toggles/webapp/toggles.ts [hunk 3]
```diff
@@ -69,6 +69,12 @@ const toggles = {
       description:
         'Adds tabbed navigation to the works page, for switching between work, item and related content',
     },
+    {
+      id: 'itemWorkLink',
+      title: 'Item page: Work page link',
+      initialValue: false,
+      description: 'Repositiions and changes the style of the work page link',
```
Comment (alexwlchan) [inline toggles/webapp/toggles.ts:76]: although it's implied by the toggle name, might be good to be explicit about which work page link:

```suggestion
      description: 'Tries an alternative style for the link from the item page to the corresponding work',
```

**Review by alexwlchan — APPROVED** [review]
a couple of v small things, but overall looks good 🥍

**Comment by gestchild** [comment]
@taceybadgerbrook we started splitting traffic 50/50 for this new link versus the old link on the 21st December

**Comment by taceybadgerbrook** [comment]
@gestchild I'm looking at the data picked up by Segment. I can't see any field with values "viewer_back_link" or "button_back_link."

image

The closest thing I can find is itemWorkLink:
image

**Comment by taceybadgerbrook** [comment]
Ok, I’m going to have a look at this in Kibana.

Data takes maybe 10 seconds before it’s visible in Kibana, so there shouldn’t be an issue with time lag.

From: Gareth Estchild ***@***.***>
Date: Wednesday, 4 January 2023 at 15:50
To: wellcomecollection/wellcomecollection.org ***@***.***>
Cc: Tacey Kobayashi ***@***.***>, Mention ***@***.***>
Subject: Re: [wellcomecollection/wellcomecollection.org] Work link (PR #9005)

Hey @taceybadgerbrook<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Ftaceybadgerbrook&data=05%7C01%7CT.Kobayashi%40wellcome.org%7C10df5b7b8bb54d0c035f08daee6b68ea%7C3b7a675a1fc84983a100cc52b7647737%7C0%7C0%7C638084442343819373%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=mFh8ntGiHHNc9tQokpvYy4oH%2FjTQQgG0maKiDVNdK%2B0%3D&reserved=0>,

The "viewer_back_link" or "button_back_link" values are in the page.query.source

Here's what I did (data for last 7 days):

  *   Filtered by page.pathname : "/item" to get total item page views = 139992
  *   Filtered by page.pathname : "/item" and properties.toggles.keyword : "itemWorkLink" to get total item page views with the toggle turned on = 15557
  *   Calculated total page item page views with toggle turned off: 139992 - 15557 = 124435
  *   Filtered by source : "button_back_link" to get total clicks on the new link = 98
  *   Filtered by source : "viewer_back_link" to get total clicks on the old link = 1594

From that we can work out:
(1594/124435) * 100 = 1.28% of people who landed on the item page with the old link clicked on it
(98/15557) * 100 = 0.63% of people who landed on the item page with the new link clicked on it

Plugging those numbers in to an online statistical significance calculator<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.investisdigital.com%2Finsights%2Fresources-and-tools%2Fstatistical-significance-calculator&data=05%7C01%7CT.Kobayashi%40wellcome.org%7C10df5b7b8bb54d0c035f08daee6b68ea%7C3b7a675a1fc84983a100cc52b7647737%7C0%7C0%7C638084442343819373%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=OFs%2FsQkOYNqlngG%2BYoe7ltckEi8JFhI8EfAT8EN%2FcIo%3D&reserved=0>:

Results
The Variant Lost

There was a -50.82% change.

This test has 100.00% statistical significance.

N.B. Things I'm not clear about.

  *   Why the traffic wasn't split 50/50 (not that is matters for the statistical significance as the numbers are high enough, but something we need to work out)
  *   I'm wondering whether the Filter by source : "button_back_link" is giving me accurate data, as I could see my own page view from today with that source if I looked for it specifically, but couldn't find it if I used that filter. Do you know if there is a delay before it would be picked up?

**Comment by taceybadgerbrook** [comment]
@gestchild at what point in the user journey do users get split 50/50? Is it when they land on an item page?

**Comment by gestchild** [comment]
Yes, when the visit an item page

**Comment by taceybadgerbrook** [comment]
@gestchild I've spotted a couple of things.

1. The definition of an item page view is page.name==work and page.pathname==/item.

2. I thought everyone who arrived at an item page view was tracked for the A/B test so shouldn't half of 139,992 served the new button and the other half the old link?
 
3. If this isn't the case and you have identified 15,557 as the users who are part of the A/B test by properties.toggles.keyword : "itemWorkLink," then we have a problem because there is a bug that Raphäelle is looking into which is causing fields such as page.name and properties.toggles to retain the same value over time. (Note: properties.toggles.keyword is the text value of properties.toggles.) I've been toggled as properties.toggles: "itemWorkLink" from the beginning.

4. Most companies will use A/B tests that use t-test or z-test methodology. These are designed for data that takes a range of values. The resulting test compares the average value of the A's against the average value of the B's. We only have 2 different values here, so a chi-squared test is more appropriate. This is designed for data that has mutually-exclusive categories of values (eg. pass/fail, yes/no.) I can't tell which Investisdigital uses, so it might be ok or we might be better off using [this one](https://www.evanmiller.org/ab-testing/chi-squared.html). Either way, we can check by putting the same values into the boxes and see if they match.

**Comment by taceybadgerbrook** [comment]
@gestchild while page.name is having problems updating, I think it's best to revisit this once the bug has been fixed.

**Pull request merged**